src/components/views/rooms/SendMessageComposer.tsx

```diff
@@ -204,7 +205,7 @@ export class SendMessageComposer extends React.Component<ISendMessageComposerPro
     }
 
     public componentDidUpdate(prevProps: ISendMessageComposerProps): void {
-        const replyingToThread = this.props.relation?.key === RelationType.Thread;
+        const replyingToThread = this.props.relation?.key === THREAD_RELATION_TYPE.name;
```

Comment:
Shouldn't this use `THREAD_RELATION_TYPE.matches(relation?.rel_type)` to support both legacy and new?